target/linux/ath79/dts/qca9563_tplink_eap245-v3.dts

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	compatible = "tplink,eap245-v3", "qca,qca9563";
+	model = "TP-Link EAP245 v3";
+
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_amber;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_amber;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_green: status_green {
+			label = "tp-link:green:status";
+			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		led_status_amber: status_amber {
+			label = "tp-link:amber:status";
+			gpios = <&gpio 9 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "factory-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot";
+				reg = <0x040000 0x040000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "partition-table";
+				reg = <0x080000 0x010000>;
+				read-only;
+			};
+
+			info: partition@90000 {
+				label = "info";
+				reg = <0x090000 0x010000>;
+				read-only;
+			};
+
+			art: partition@a0000 {
+				label = "art";
+				reg = <0x0a0000 0x010000>;
+				read-only;
+			};
+
+			partition@b0000 {
+				label = "extra-para";
+				reg = <0x0b0000 0x010000>;
+				read-only;
+			};
+
+			partition@c0000 {
+				compatible = "openwrt,elf";
+				label = "firmware";
+				reg = <0x0c0000 0xe40000>;
+			};
+
+			partition@f00000 {
+				label = "config";
+				reg = <0xf00000 0x030000>;
+				read-only;
+			};
+
+			partition@f30000 {
+				label = "mutil-log";
+				reg = <0xf30000 0x080000>;
+				read-only;
+			};
+
+			partition@fb0000 {
+				label = "oops";
+				reg = <0xfb0000 0x040000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0x1>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "sgmii";
```

Comment:
@blocktrron Reading the sources for the ar71xx ethernet drive, I suspect that specifying phy-mode in this node is not useful since it is never really used. The phy-mode at line 153 does appear to be used by the driver. Looking at other devices, I couldn't really see a specific pattern in where to use phy-mode. Would you be able to clarify?

Reply:
Isn't that used in the mac layer?
There is not much documentation about it. :/
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/net/ethernet-controller.yaml#n66

Reply:
I don't remember finding that document when I looked into this earlier. Thinking about it, it makes sense though. Both the mac and the phy have to use the same link type, so I has to be defined for both (if I understand this correctly).